add a request-store and put the settings_max_updated_on into the request to avoid redundant calls

mugwump opened the pull request:
Implements ticket [2319](https://www.openproject.org/issues/2319)

The Request-Store is a middleware allows to store thread-local-variables, that are properly cleaned up after the request. This mechanism is then used to store the Setting.maximun(:updated_on) in a thread-local variable, to avoid redundant db-calls/cache-calls. 

PS: This only gets rid of the updated-on calls, a longer-term fix would probably be to store accessed setting in a thread-local or even read the whole setting and store it... 

PPS: This is largely pulled from https://github.com/steveklabnik/request_store, see the readme for a lengthy discussion about thread-local variables.

mugwump committed:
Stores the Setting.maximum(:updated_on) in a thread-local-variable

this gets rid of numerous redundant db-calls to check with every call to
a setting, if the settings have been updated. Checking this one and one
time only for a request should be sufficient.

let's see, if this works for the admin-specs to update settings.

mugwump committed:
Added the request-store, to make sure that thread-local variables are cleaned up after the request

apfelfabrik commented:
Why not just https://github.com/steveklabnik/request_store then?
Also, the commit message does not state where the code is largely pulled from.

Using the gem directly would also fix the middleware class having no copyright header :-)
And how about a changelog entry?

mugwump committed:
replaced the pulled code with proper gem-reference

The railtie now automatically makes sure, that the middleware is used,
and the included gem better respects the copyright of the used solution.

mugwump committed:
forgot the removal of the middleware&the comment in application.rb

apfelfabrik commented:
~~`ApplicationController with log_requesting_user enabled.ApplicationController with log_requesting_user enabled should log the current user` consistently fails on postgres for me. Letting it run a few more times after cleaning the database.~~ All is good.

apfelfabrik commented:
What about the changelog/ticket?

apfelfabrik committed:
Changelog.
